### DISCUSSION TITLE
Pros and cons of various payment methods

### DISCUSSION THREAD

**Discussion by willhansen** [openfare/openfare#8] [General]
There aren't a lot of payment methods that support the use case targeted by OpenFare (small payments split hundreds to thousands of ways).  I'm thinking this would be a good place to discuss which ones are viable to implement and use.

- Paypal
  - Pros: 
    - Familiar to many users
    - Commonly supported
  - Cons:
    - Transaction fees have a percentage and fixed part.  
      - The fixed part adds up quick with many contributors.  0.30$ * 1000 contributors = 300$
    - Known to freeze accounts sometimes (especially with proximity to cryptocurrencies)
    - Batch payments require a business account
- Bitcoin lightning network
  - Pros:
    - Low transaction fees
    - Fast
  - Cons: 
    - Requires a bitcoin transaction to get currency on and off the lightning network (expensive and slow)
    - The technology is not even ready yet? (lnpay?)
- Other crypto-currencies
  - Pros: 
    - More established than the lightning network
    - Low fees and delays (depending on which currency)
  - Cons:
    - Extremely volatile coin values.  
      - Makes listing prices in coins non-viable.  
      - Fetching current coin-to-USD value at time of payment could lead to payers attempting to time the market?
    - Onboarding and offboarding money must go through crypto-currency exchanges
    - How to handle a lock file payment method for a currency that no longer exists?
      - (Only applicable for unmaintained code)
      - Impossible to pay, so the associated code can only be re-written?
      - Have a more reliable secondary payment method listed (like paypal, or mailing a cheque) just in case?

**Comment by rndhouse** [comment]
I've just created a GitHub "Discussion" page for this repository: https://github.com/openfare/openfare/discussions

I'm going to make this issue a thread in that section. I hope you don't mind. I can see this discussion being more open ended than what an issue should handle.

**Reply by willhansen** [reply]
@rndhouse How far have you planned out the payment methods of OpenFare?

1. Is the Bitcoin lightning network in a usable state in general?
  - Is there a way around requiring bitcoin transactions to use any lightning channel?
2. Is there a way around paypal fees that cripple one-to-many transactions without having a business account on the sending side for batch payments?
  - Are there plans to do something along the lines of a one-to-one transaction from a user to an OpenFare business entity, then use batch payments to contributors?  If so,  how much more complicated does that make things legally/bureaucratically?
3. Do you think using other crypto-currencies could be viable?
  - It's probably a lost cause to have a user set up a wallet in some obscure cryptocurrency they've never heard of, so are you familiar with any "credit card to cryptocurrency to one-to-many transaction, all in one payment gateway" type things?

**Comment by willhansen** [comment]
Based on looking at the top cryptocurrencies, my findings are that Bitcoin, Ethereum, Cardano, and the stable-coins have too-high fees, and Solana briefly went down entirely because they got overloaded.  That leaves Ripple coin (XRP) as the top contender.
